src/libraries/Microsoft.PowerFx.Core/Logging/Trackers/DelegationTelemetryInfo.cs

```diff
@@ -81,8 +82,36 @@ public static DelegationTelemetryInfo CreateImpureNodeTelemetryInfo(TexlNode nod
                     var funcName = binding?.GetInfo(callNode)?.Function?.Name ?? string.Empty;
                     return new DelegationTelemetryInfo(funcName);
                 default:
-                    return new DelegationTelemetryInfo(node.ToString());
+                    return new DelegationTelemetryInfo(StructuralPrint.Print(node, binding));
             }
         }
+
+        public static DelegationTelemetryInfo CreateAsyncNodeTelemetryInfo(TexlNode node, TexlBinding binding = null)
+        {
+            Contracts.AssertValue(node);
+            Contracts.AssertValueOrNull(binding);
+
+            switch (node.Kind)
+            {
+                case NodeKind.Call:
+                    var callNode = node.AsCall();
+                    var funcName = binding?.GetInfo(callNode)?.Function?.Name ?? string.Empty;
+                    return new DelegationTelemetryInfo(funcName);
+                default:
+                    return new DelegationTelemetryInfo(StructuralPrint.Print(node, binding));
+            }
+        }
+
+        public static DelegationTelemetryInfo CreateUnsupportArgTelmetryInfo(DType dType)
```

Comment:
Nit: Should be "Telemetry" here and below

Reply:
Corrected